src/benchmarks/gc/GC.Infrastructure/GC.Infrastructure/Commands/ASPNetBenchmarks/AspNetBenchmarksCommand.cs

```diff
@@ -110,116 +266,31 @@ public static AspNetBenchmarkResults RunASPNetBenchmarks(ASPNetBenchmarksConfigu
             // For each benchmark, iterate over all specified runs.
             foreach (var c in benchmarkToNameCommandAsKvpList)
             {
-                foreach (var run in configuration.Runs)
+                foreach (var run in configuration.Runs!)
                 {
-                    OS os = !c.Key.Contains("Win") ? OS.Linux : OS.Windows; 
-                    (string, string) commandLine = ASPNetBenchmarksCommandBuilder.Build(configuration, run, c, os);
+                    const string NON_RESPONSIVE = @"for 'application' is invalid or not responsive: ""No such host is known";
+                    const string TIME_OUT = "[Time Out]";
+                    ProcessExecutionDetails result = ExecuteBenchmarkForRun(configuration, run, c);
+                    string key = GetKey(c.Key, run.Key);
 
-                    string outputPath = Path.Combine(configuration.Output.Path, run.Key);
-                    if (!Directory.Exists(outputPath))
-                    {
-                        Directory.CreateDirectory(outputPath);
-                    }
+                    bool timeout = result.StandardError.Contains(TIME_OUT);
+                    bool nonResponsive = result.StandardOut.Contains(NON_RESPONSIVE) || result.StandardError.Contains(NON_RESPONSIVE);
+                    bool timeoutOrNonResponsive = timeout || nonResponsive;
 
-                    // There are 3 main ASP.NET errors:
-                    // 1. The server is unavailable - this could be because you aren't connected to CorpNet or the machine is down.
-                    // 2. The crank commands are incorrect.
-                    // 3. Test fails because of a test error.
-
-                    // Launch new crank process.
-                    int exitCode = -1;
-                    string logfileOutput = Path.Combine(outputPath, $"{GetKey(c.Key, run.Key)}.log");
-                    StringBuilder output = new();
-                    StringBuilder error = new();
-
-                    using (Process crankProcess = new())
+                    // Wait 2 minutes and then retry if the run timed out or the host was non-responsive (post corp-net connection check).
```

Comment:
Another case is the VPN dropping in the middle of a run, which unfortunately seems to be common.

Of course, the run is toast at that point, so it doesn't matter much.  This will cause each test post-drop to run twice, but I don't think that makes anything worse than it was.  I guess a script running ASP.NET benchmarks and then something else would have to wait a bit longer.  I think this PR should go in without addressing this.  (And the only way seems to be to reconnect the VPN, if that is even possible to do programmatically?)

Reply:
Yeah, agreed - I think we are out of luck if we run into the aforementioned situation. The recommendation here is to run the benchmarks on the internal GC machines where this state will never arise.